src/stores/TierStore.js

```diff
@@ -168,33 +168,60 @@ class TierStore {
   }
 
   @action addWhitelistItem = ({ addr, min, max }, crowdsaleNum) => {
-    const tier = this.tiers[crowdsaleNum]
+    const { whitelist } = this.tiers[crowdsaleNum]
+    const newItem = { addr, min, max }
+    const _addr = addr.toLowerCase()
+    const isAdded = whitelist.find(item => item.addr.toLowerCase() === _addr)
 
-    const whitelist = tier.whitelist.slice()
+    if (this.deployedContract) {
+      const storedIndex = whitelist.findIndex(item => item.addr.toLowerCase() === _addr && item.stored)
+      const duplicatedIndex = whitelist.findIndex(item => item.addr.toLowerCase() === _addr && !item.stored && item.duplicated)
 
-    const isAdded = whitelist.find(item => item.addr === addr && !item.deleted)
+      if (duplicatedIndex > -1) return
 
-    if (isAdded) return
+      if (storedIndex > -1) {
+        whitelist[storedIndex].duplicated = true
+        newItem.duplicated = true
 
-    const whitelistElements = tier.whitelistElements.slice()
-    const whitelistNum = whitelistElements.length
+      } else if (isAdded) return
 
-    whitelistElements.push({ addr, min, max, whitelistNum, crowdsaleNum })
-    whitelist.push({ addr, min, max })
+    } else if (isAdded) return
 
-    this.setTierProperty(whitelistElements, 'whitelistElements', crowdsaleNum)
-    this.setTierProperty(whitelist, 'whitelist', crowdsaleNum)
+    whitelist.push(newItem)
+    this.sortWhitelist(crowdsaleNum)
+  }
+
+  @action sortWhitelist = (crowdsaleNum) => {
+    this.tiers[crowdsaleNum].whitelist = this.tiers[crowdsaleNum].whitelist.sort((prev, curr) => {
+      const currentAddress = curr.addr.toLowerCase()
+      const previousAddress = prev.addr.toLowerCase()
+
+      return currentAddress > previousAddress ? -1 : currentAddress === previousAddress ? curr.stored ? 1 : -1 : 1
+    })
   }
 
   @action removeWhitelistItem = (whitelistNum, crowdsaleNum) => {
-    let whitelist = this.tiers[crowdsaleNum].whitelist.slice()
-    whitelist[whitelistNum].deleted = true
-    this.setTierProperty(whitelist, 'whitelist', crowdsaleNum)
+    const removedItem = this.tiers[crowdsaleNum].whitelist.splice(whitelistNum, 1)[0]
```

Comment:
This needs tests too.

Reply:
done, and added test for the other modified/created methods